File: backend/app/schemas/appointments.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class AppointmentBase(BaseModel):
    service_id: str
    client_name: str
    client_email: str
    start_time: datetime
    end_time: datetime


class AppointmentCreate(AppointmentBase):
    pass


class AppointmentUpdate(BaseModel):
    client_name: Optional[str] = None
    client_email: Optional[str] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None
    status: Optional[str] = None


class AppointmentResponse(AppointmentBase):
    id: str
    status: str
    stripe_payment_intent_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class TimeSlot(BaseModel):
    start: datetime
    end: datetime
    available: bool


class PublicProfile(BaseModel):
    id: str
    full_name: Optional[str] = None
    avatar_url: Optional[str] = None
    public_slug: str

File: frontend/src/types/appointments.ts
/**
 * Tipagens para o domínio de Agendamentos.
 *
 * Alinhado com o schema backend: AppointmentResponse.
 */

export type AppointmentStatus = 'pending' | 'pending_payment' | 'confirmed' | 'canceled' | 'cancelled' | 'completed';

export interface Appointment {
  id: string;
  professional_id: string;
  service_id: string;
  student_id?: string;
  client_name?: string;
  client_email?: string;
  start_time: string;   // ISO 8601 UTC
  end_time: string;     // ISO 8601 UTC
  status: AppointmentStatus;
  google_event_id?: string;
  created_at: string;
  updated_at: string;
}

export interface AppointmentCreate {
  professional_id: string;
  service_id: string;
  student_name: string;
  student_email: string;
  student_phone?: string;
  start_time: string;
  end_time: string;
}

export interface PublicProfile {
  id: string;
  full_name: string;
  avatar_url?: string;
  public_slug: string;
}

export interface TimeSlot {
  start: string;
  end: string;
  available: boolean;
}